flink-ml-lib/src/test/java/org/apache/flink/ml/classification/NaiveBayesTest.java

```diff
@@ -0,0 +1,349 @@
+                    Row.of(1., Vectors.dense(1, 1., 0., 1.), 11.0),
+                    Row.of(1., Vectors.dense(2, 0., 1., 1.), 11.0),
+                    Row.of(1., Vectors.dense(2, 0., 1., 1.5), 11.0),
+                    Row.of(2., Vectors.dense(3, 1.5, 1., 0.5, 3.), 10.0),
+                    Row.of(1., Vectors.dense(1, 1., 1.5, 0., 1.), 10.0),
+                    Row.of(2., Vectors.dense(4, 1., 1., 0., 1.), 10.0)
+                };
+        runAndCheck();
+    }
+
+    @Test
+    public void testSaveLoad() throws Exception {
+        errorMessage = "Naive Bayes should be able to save Model to filesystem and load correctly.";
+        isSaveLoad = true;
+        runAndCheck();
+    }
+
+    @Test
+    public void testGetModelData() throws Exception {
+        errorMessage =
+                "Naive Bayes should be able to provide model data with getModelData() method.";
+        trainData =
+                new Row[] {
+                    Row.of(1., Vectors.dense(1, 1.), 11.0), Row.of(2., Vectors.dense(2, 1.), 11.0)
+                };
+        NaiveBayesModel model = getModel();
+        NaiveBayesModelData expected =
+                new NaiveBayesModelData(
+                        new HashMap[][] {
+                            new HashMap[] {
+                                new HashMap<Double, Double>() {
+                                    {
+                                        put(2.0, -0.6931471805599453);
+                                        put(1.0, -0.6931471805599453);
+                                    }
+                                },
+                                new HashMap<Double, Double>() {
+                                    {
+                                        put(1.0, 0.0);
+                                    }
+                                },
+                            },
+                        },
+                        new double[] {0.0},
+                        new double[] {11.0});
+        NaiveBayesModelData actual =
+                NaiveBayesModelData.toDataStream(tEnv, model.getModelData()[0])
+                        .executeAndCollect()
+                        .next();
+        assertArrayEquals(errorMessage, expected.labels, actual.labels, 1e-5);
+        assertArrayEquals(errorMessage, expected.piArray, actual.piArray, 1e-5);
+        assertArrayEquals(errorMessage, expected.theta[0], actual.theta[0]);
+    }
+
+    @Test
+    public void testSetModelData() throws Exception {
+        errorMessage =
+                "Naive Bayes should be able to provide model data with getModelData() method.";
+
+        NaiveBayesModel modelA = getModel();
+
+        Table modelData = modelA.getModelData()[0];
+        NaiveBayesModel modelB = new NaiveBayesModel().setModelData(modelData);
+        ReadWriteUtils.updateExistingParams(modelB, modelA.getParamMap());
+
+        checkResult(modelB);
+    }
+
+    private void runAndCheck() throws Exception {
```

Comment:
It is probably preferred to setup a best practice of writing unit tests for algorithms.

Any chance we can re-use the test pattern (e.g. `executeAndCollect`) used in `KMeansTest`?